src/main/ruby/core/posix.rb

```diff
@@ -136,6 +136,8 @@ def self.attach_function_eagerly(native_name, argument_types, return_type,
           end
         elsif return_type == :pointer
           result = Truffle::FFI::Pointer.new(Truffle::Interop.as_pointer(result))
+        elsif return_type == :ssize_t && Truffle.invoke_primitive(:fixnum_fits_into_int, result)
```

Comment:
The explicit check `fixnum_fits_into_int` is unneeded, `FixnumOperations.lower` already does it and handle if it doesn't fit.
Can you remove it?